genoml/preprocessing/munging.py

```diff
@@ -15,340 +15,369 @@
+                    f"{col} is likely a binary indicator or a proportion and will not "
+                    "be scaled, just + 1 all the values of this variable and rerun to "
+                    "flag this column to be scaled.",
+                )
+            else:
+                addit_df[col] = (addit_df[col] - addit_df[col].mean()) / addit_df[
+                    col
+                ].std(ddof=0)
+
+        print(
+            "\n"
+            "You have just Z-scaled your non-genotype features, putting everything on "
+            "a numeric scale similar to genotypes.\n"
+            "Now your non-genotype features might look a little closer to zero "
+            "(showing the first few lines of the left-most and right-most columns)..."
+        )
+        print("#" * 70)
+        print(addit_df.describe())
+        print("#" * 70)
+        return addit_df
+
+    def harmonize_refs(self, merged_df: pd.DataFrame) -> pd.DataFrame:
+        """Harmonizes data columns with an external reference file.
+
+        > Checking the reference column names flag
+        > If this is a step that comes after harmonize, then a .txt file with columns
+        > to keep should have been produced. This is a list of column names from the
+        > reference dataset that the test dataset was harmonized against. We want to
+        > compare apples to apples, so we will only keep the column names that match.
+        """
+        if not self.refColsHarmonize:
+            return merged_df
+        print(
+            "\n"
+            f"Looks like you are munging after the harmonization step. Great! We will "
+            f"keep the columns generated from your reference dataset from that "
+            f"harmonize step that was exported to this file: {self.refColsHarmonize}\n"
+        )
+        with open(self.refColsHarmonize, "r") as refCols_file:
+            ref_column_names_list = refCols_file.read().splitlines()
+
+        # Keep the reference columns from the test dataset if found in test data
+        matching_cols = merged_df[
+            np.intersect1d(merged_df.columns, ref_column_names_list)
+        ]
+
+        # Make a list of final features that will be included in the model
+        # This will be used again when remunging the reference dataset
+        matching_cols_list = matching_cols.columns.values.tolist()
+
+        # Save out the final list
+        intersecting_cols_outfile = f"{self.run_prefix}.finalHarmonizedCols_toKeep.txt"
+
+        with open(intersecting_cols_outfile, "w") as f:
+            for col in matching_cols_list:
+                f.write(f"{col}\n")
+
+        print(
+            "A final list of harmonized columns between your reference and test "
+            f"dataset has been generated here: {intersecting_cols_outfile}\n"
+            "Use this to re-train your reference dataset in order to move on to "
+            "testing."
+        )
+        return matching_cols
+
+
+def get_bash_scripts(
```

Comment:
Since this one just returns a list of `str`s, I left it not private for users to check later.

I am also not 100% what each of these plink functions does so I didnt alter it. Renamed to `get_plink_bash_scripts`.

Reply:
Sounds good!